Calculator_program.c

```diff
@@ -0,0 +1,127 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#define MAX_LEN 100 // Maximum length
+
+// Function to calculate expressions 
+float calculateExpression(char *expr) {
+    float values[MAX_LEN]; 
+    char operators[MAX_LEN]; 
+    int valTop = -1; 
+    int opTop = -1; 
+
+    int i = 0;
+    //Function to perform operations
+    float applyOperation(float a, float b, char op) {
+        if (op == '+') return a + b;
+        if (op == '-') return a - b;
+        if (op == '*') return a * b;
+        if (op == '/') {
+            if (b == 0) {
+                printf("Division by zero is not allowed.\n");
+                exit(1);

```

Comment:
Try to modify the code such that there is no need to use exit()

Reply:
👍